rfcs/0000-rlp-status.md

```diff
@@ -0,0 +1,157 @@
+|            | False  | "PolicyNotEnforced"   | "RateLimitPolicy has encountered an error"         |           |
+
+
+A simplified version and more aligned with the Kubernetes objects implementation could be represented as the following.
+
+**Conditions**
+
+All conditions are top-level.
+
+| Type        | Status | Reason                      | Message                                                    |
+|-------------|--------|-----------------------------|------------------------------------------------------------|
+| Progressing | True   | "PolicyCreated"             | "RateLimitPolicy created"                                  |
+|             | True   | "PolicyUpdated"             | "RateLimitPolicy has been updated"                         |
+|             | True   | "PolicyApplied"             | "RateLimitPolicy has been successfully applied             |
+|             | True   | "PolicyReconciled"          | "RateLimitPolicy has been successfully reconciled"         |
+|             | False  | "PolicyEnforced"            | "RateLimitPolicy has been successfully enforced"           |
+|             | False  | "PolicyError"               | "RateLimitPolicy has encountered an error"                 |
+| Protected   | True   | "PolicyEnforced"            | "RateLimitPolicy has been successfully enforced"           |
+|             | False  | "PolicyNotEnforced"         | "RateLimitPolicy has encountered an error while enforcing" |
+| Failed      | True   | "PolicyValidationError"     | "RateLimitPolicy has failed to validate"                   |
+|             | True   | "PolicyReconciliationError" | "RateLimitPolicy has encountered a reconciliation error"   |
+|             | True   | "PolicyServiceError"        | "RateLimitPolicy has encountered has failed to enforce"    |
+|             | False  | "PolicyEnforced"            | "RateLimitPolicy has been successfully enforced"           |
+
+
+### Notes
+* The messages, the ones corresponding to the _falsey status_, might reflect the error that encountered.
+* It's possible to have the _Failed_ state as a top level condition too. In this case might be useful to add a third 
+"Unknown" status.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- This proposal will require to change the code controllers assert the status
+- Since the Status is part of the "API", won't be backwards compatible
+- Documentation updating
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+-
+
+# Prior art
+[prior-art]: #prior-art
+
+* [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status)
+* [Current RateLimitPolicy Status work](https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/ratelimitpolicy_status.go)
+
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Should this proposal extend to the KAP as well?
```

Comment:
I think where possible if we can align the APIs it would make sense from a user exp point of view. Also interesting to think about KAP in the context of `protected` `unprotected`